src/coreclr/jit/helperexpansion.cpp

```diff
@@ -697,7 +701,7 @@ PhaseStatus Compiler::fgExpandHelper(bool skipRarelyRunBlocks)
         }
 
         // Expand and visit the last block again to find more candidates
-        while (fgExpandHelperForBlock<ExpansionFunction>(block))
+        while (fgExpandHelperForBlock<ExpansionFunction>(&block))
```

Comment:
I would make sure that we indeed get an updated block if `fgExpandHelperForBlock()` returned `true.`

```c++
INDEBUG(BasicBlock* origBlock = block);
while (fgExpandHelperForBlock<ExpansionFunction>(&block))
{
   result = PhaseStatus::MODIFIED_EVERYTHING; 
#ifdef DEBUG
   assert(origBlock != block);
   origBlock = block;   
#endif
}
```